mono/metadata/w32handle.c

```diff
@@ -96,8 +237,10 @@ mono_w32handle_lookup_data (gpointer handle, MonoW32HandleBase **handle_data)
 		return FALSE;
 
 	offset = SLOT_OFFSET ((gsize) handle);
-	if (private_handles [index][offset].type == MONO_W32HANDLE_UNUSED)
+	if (private_handles [index][offset].type == MONO_W32HANDLE_UNUSED) {
+		assert_handle_is_dead (handle);
```

Comment:
Isn't this racing ? Since you do not atomically check the type and the state, it might be possible that the handle is initialized right between these 2 calls.